pyximport/pyximport.py

```diff
@@ -1,606 +1,11 @@
-
-    :param load_py_module_on_import_failure: If the compilation of a ``.py``
-        file succeeds, but the subsequent import fails for some reason,
-        retry the import with the normal ``.py`` module instead of the
-        compiled module.  Note that this may lead to unpredictable results
-        for modules that change the system state during their import, as
-        the second import will rerun these modifications in whatever state
-        the system was left after the import of the compiled module
-        failed.
-
-    :param inplace: Install the compiled module
-        (``.so`` for Linux and Mac / ``.pyd`` for Windows)
-        next to the source file.
-
-    :param language_level: The source language level to use: 2 or 3.
-        The default is to use the language level of the current Python
-        runtime for .py files and Py2 for ``.pyx`` files.
-    """
-    if setup_args is None:
-        setup_args = {}
-    if not build_dir:
-        build_dir = os.path.join(os.path.expanduser('~'), '.pyxbld')
-
-    global pyxargs
-    pyxargs = PyxArgs()  #$pycheck_no
-    pyxargs.build_dir = build_dir
-    pyxargs.build_in_temp = build_in_temp
-    pyxargs.setup_args = (setup_args or {}).copy()
-    pyxargs.reload_support = reload_support
-    pyxargs.load_py_module_on_import_failure = load_py_module_on_import_failure
-
-    has_py_importer, has_pyx_importer = _have_importers()
-    py_importer, pyx_importer = None, None
-
-    if pyimport and not has_py_importer:
-        py_importer = PyImporter(pyxbuild_dir=build_dir, inplace=inplace,
-                                 language_level=language_level)
-        # make sure we import Cython before we install the import hook
-        import Cython.Compiler.Main, Cython.Compiler.Pipeline, Cython.Compiler.Optimize
-        sys.meta_path.insert(0, py_importer)
-
-    if pyximport and not has_pyx_importer:
-        pyx_importer = PyxImporter(pyxbuild_dir=build_dir, inplace=inplace,
-                                   language_level=language_level)
-        sys.meta_path.append(pyx_importer)
-
-    return py_importer, pyx_importer
-
-
-def uninstall(py_importer, pyx_importer):
-    """
-    Uninstall an import hook.
-    """
-    try:
-        sys.meta_path.remove(py_importer)
-    except ValueError:
-        pass
-
-    try:
-        sys.meta_path.remove(pyx_importer)
-    except ValueError:
-        pass
-
-
-# MAIN
-
-def show_docs():
-    import __main__
-    __main__.__name__ = mod_name
-    for name in dir(__main__):
-        item = getattr(__main__, name)
-        try:
-            setattr(item, "__module__", mod_name)
-        except (AttributeError, TypeError):
-            pass
-    help(__main__)
 
+if sys.version_info.major == 3 and sys.version_info.minor >= 5:
+    # _pyximport3 module requires at least Python 3.5
+    from pyximport._pyximport3 import install, uninstall, show_docs
+else:
+    from pyximport._pyximport2 import install, uninstall, show_docs
```

Comment:
Let's not write code that we already know will break in Python 4.
```suggestion
if sys.version_info < (3, 5):
    # _pyximport3 module requires at least Python 3.5
    from pyximport._pyximport2 import install, uninstall, show_docs
else:
    from pyximport._pyximport3 import install, uninstall, show_docs
```